eZ/Publish/Core/Repository/SectionService.php

```diff
@@ -279,7 +282,10 @@ public function assignSection(ContentInfo $contentInfo, Section $section)
         $loadedContentInfo = $this->repository->getContentService()->loadContentInfo($contentInfo->id);
         $loadedSection = $this->loadSection($section->id);
 
-        if ($this->repository->canUser('section', 'assign', $loadedContentInfo, $loadedSection) !== true) {
+        // REVIEW NOTE: NewSection limitation expects target to be the section, while Role Subtree limitations will abstain from checking
+        // due to target being section. We could consider changing Subtree to load targets from content and check anyway. So
+        // we make sure section assignments are limited by subtree limitations which could make sense given object is content.
+        if (!$this->permissionResolver->canUser('section', 'assign', $loadedContentInfo, [$section])) {
```

Comment:
WDYT? :)

Reply:
**Question 1**:
>  REVIEW NOTE: NewSection limitation expects target to be the section, while Role Subtree limitations will abstain from checking due to target being section.

So you're saying that if we have Subtree limitation, here we can assign section to content which belongs to another subtree? :)

**Question 2**: 
any reason you've replaced `$loadedSection` with `$section` in this line?

Reply:
> any reason you've replaced $loadedSection with $section in this line?

no, that is a "mistake" and I'll change it back to use `$loadedSection . _Even if I perosnally don't think we need to re-load things like this in the first place ;)_

Reply:
> I'm saying both NewSection and SubtreeLimitation (role limitation) could be adhered to.
> I can ope that as separate PR to master so we can take that as separate thing?

Yes please :)